freediscovery/interpretation.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import re
+import matplotlib as mpl
+import matplotlib.cm as cm

```

Comment:
all matplotlib imports should go inside `_make_cmap` to make sure we can import this module without matplotlib...